packages/driver/src/cy/ensures.ts

```diff
@@ -383,6 +384,32 @@ export const create = (state: StateFunc, expect: $Cy['expect']) => {
     })
   }
 
+  /**
+   * ensureCommandCanCommunicateWithAUT will check if the command should be able to communicate with the AUT
+   * If we can not communicate, throw an error.
+   * Intended to use within retry loops.
+   * err: optional error to pass end to be appended to if the assertion happened while the aut was cross origin.
+   * @returns true or throws an error
+   */
+  const ensureCommandCanCommunicateWithAUT = (err?): boolean => {
+    if (!isRunnerAbleToCommunicateWithAut()) {
+      const crossOriginCommandError = $errUtils.errByPath('miscellaneous.cross_origin_command', {
+        commandOrigin: window.location.origin,
+        autOrigin: Cypress.state('autLocation').originPolicy,
```

Comment:
This looks like it's using the global `Cypress`. While that's technically possible since it's defined globally, I think it's better to pass `Cypress` into the `create` function that closes over this method, then [pass `Cypress` in when invoking it](https://github.com/cypress-io/cypress/blob/eb5860e7830c438f159d365daae8544b67e15e65/packages/driver/src/cypress/cy.ts#L332).

Reply:
good call, state is actually already passed in on create, so i can just use that directly.